relay-spans/src/v2_to_v1.rs

```diff
@@ -0,0 +1,794 @@
+use std::str::FromStr;
+
+use relay_event_schema::protocol::{SpanKind, SpanV2Link};
+
+use crate::status_codes;
+use relay_event_schema::protocol::{
+    EventId, Span as SpanV1, SpanData, SpanId, SpanLink, SpanStatus, SpanV2, SpanV2Kind,
+    SpanV2Status,
+};
+use relay_protocol::{Annotated, FromValue, Object, Value};
+
+#[allow(dead_code)]
+/// Transforms a Sentry Span V2 to a Sentry Span.
+pub fn span_v2_to_span_v1(span_v2: SpanV2) -> SpanV1 {
+    let mut exclusive_time_ms = 0f64;
+    let mut data = Object::new();
+    let SpanV2 {
+        start_timestamp,
+        end_timestamp,
+        trace_id,
+        span_id,
+        parent_span_id,
+        name,
+        kind,
+        links,
+        attributes,
+        status,
+        is_remote,
+        ..
+    } = span_v2;
+
+    let mut op = name;
+    let mut description = Annotated::empty();
+    let mut http_method = Annotated::empty();
+    let mut http_route = Annotated::empty();
+    let mut http_status_code = Annotated::empty();
+    let mut grpc_status_code = Annotated::empty();
+    let mut platform = Annotated::empty();
+    let mut segment_id = Annotated::empty();
+    let mut profile_id = Annotated::empty();
+
+    if let Some(attributes) = attributes.into_value() {
+        for (key, attribute) in attributes.into_iter() {
+            if let Some(value) = attribute.into_value().map(|v| v.value.value) {
```

Comment:
Should be able to collapse this into a single iterator, maybe nicer than 3 indentation levels.